lib/excon/middlewares/redirect_follower.rb

```diff
@@ -11,9 +23,15 @@ def get_header(datum, header)
       end
 
       def response_call(datum)
+        if datum[:redirects_remaining] <= 0
```

Comment:
I feel like we may want to raise an error in this case? The idempotent middleware effectively does this (by allowing the actual error to escape), and I would imagine if you run out of redirects and still have more left, you probably want to know in an exceptional way?

Reply:
Makes sense - updated